x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/utils/NativeMemoryCalculator.java

```diff
@@ -107,18 +150,11 @@ private static long allowedBytesForMl(long machineMemory, Long jvmSize, int maxM
             if (machineMemory - jvmSize < OS_OVERHEAD || machineMemory == 0) {
                 return machineMemory / 100;
             }
-            // This calculation is dynamic and designed to maximally take advantage of the underlying machine for machine learning
-            // We only allow 200MB for the Operating system itself and take up to 90% of the underlying native memory left
-            // Example calculations:
-            // 1GB node -> 41%
-            // 2GB node -> 66%
-            // 16GB node -> 87%
-            // 64GB node -> 90%
             long memoryPercent = Math.min(90, (int)Math.ceil(((machineMemory - jvmSize - OS_OVERHEAD) / (double)machineMemory) * 100.0D));
-            return machineMemory * memoryPercent / 100;
+            return (long)(machineMemory * (memoryPercent / 100.0));
```

Comment:
Just to confirm, is this changed purely for clarity?  I don't think the new formula actually changes the result unless a machine has more than 80 petabytes of RAM?  (And at that level converting to double will lose precision so may introduce subtle rounding issues.)

Reply:
Clarity. 

All these calculations are already hard enough to grok. I am trying to make them as clear as possible